util/kvdb-rocksdb/src/lib.rs

```diff
@@ -510,17 +512,25 @@ impl Database {
 
 	/// Get database iterator for flushed data.
 	pub fn iter(&self, col: Option<u32>) -> Option<DatabaseIterator> {
-		//TODO: iterate over overlay
 		match *self.db.read() {
 			Some(DBAndColumns { ref db, ref cfs }) => {
+				let overlay = &self.overlay.read()[Self::to_overlay_column(col)];
+				let overlay_data = overlay.iter()
+					.filter_map(|(k, v)| match v {
+						&KeyState::Insert(ref value) |
+						&KeyState::InsertCompressed(ref value) =>
+							Some((k.clone().into_vec().into_boxed_slice(), value.clone().into_vec().into_boxed_slice())),
+						&KeyState::Delete => None,
+					}).collect::<Vec<_>>();
```

Comment:
Do we really need to allocate here?

Reply:
I'm having trouble finding some other way to convert `elasticarray`'s to boxed slices.